### PR TITLE
Fixing of the contrast of the borders "Popup" and "Standard" buttons

### LINKED ISSUE #6021: The color contrast of the button border is less than 3:1
.NET Core Version: 
- 7.0.0-alpha.1.21519.1

Have you experienced this same bug with .NET Framework?: 
- Yes (based on [#1101503](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1101503))

**Repro steps:**

1. Create a form with a button.
2. Run the application.

**Actual behavior:**
`Button.FlatStyle= FlatStyle.Popup`:

image

`Button.FlatStyle= FlatStyle.Standard`:

image

`Button.FlatStyle= FlatStyle.System`:
image

**Expected behavior:**
Luminosity contrast ratio of the button border is greater than 3:1.

### PR DESCRIPTION
**Pull request by SergeySmirnov-Akvelon**
Fixes #6021

## Proposed changes
- The problem is reproduced because the default colors are too low contrast.
- As a fix for the `Popup` button, added logic to darken the current color.
- As a fix for the `Standard` button, added logic for drawing the border with the necessary contrast.

**Notes:**
`Flat` buttons do not need to be fixed as they already have the necessary contrast ratio:

image

The fix does not affect the `System` button, since the Windows team is responsible for it rendering

## Customer Impact
<details>
<summary>"Popup" button</summary>

**Before fix:**
image

**After fix:**
image
</details>

<details>
<summary>"Popup" button in the "Hovered" state</summary>

**Before fix:**
image

**After fix:**
image
</details>

<details>
<summary>"Standard" button</summary>

**Before fix:**
image

**After fix:**
image
</details>

## Regression? 
- No

## Risk
- Minimal

## Test methodology 
- CTI team

## Accessibility testing  
- Accessibility Insights

## Test environment(s) 
- Microsoft Windows [Version 10.0.19041.804]
- .NET Core SDK: 7.0.0-alpha.1.21522.9

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6059)

### CODE REVIEW

**Review by vladimir-krestov — DISMISSED** [review]
LGTM ✔️

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ internal abstract partial class ButtonBaseAdapter
         // SystemInformation.Border3DSize + 2 pixels for focus rect
         protected const int ButtonBorderSize = 4;
 
+        protected const float ButtonBorderDarkerOffset = 0.8f;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs:21]: In general, when you add constant like this, Please add a comment on how you come to this value.

Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs:22]: Thank you. Added comment

**Review by dreddy-work — APPROVED** [review]
LGTM.

**Review by vladimir-krestov — APPROVED** [review]

**Comment by SergeySmirnov-Akvelon** [comment]
Approved by CTI team

File: src/System.Windows.Forms/src/System/Windows/Forms/ControlPaint.cs [hunk 2]
```diff
@@ -308,11 +308,14 @@ public unsafe static IntPtr CreateHBitmapTransparencyMask(Bitmap bitmap)
             }
         }
 
-        /// <summary>
-        ///  Creates a Win32 HBITMAP out of the image. You are responsible for deleting the HBITMAP. If the image uses
-        ///  transparency, the background will be filled with the specified color.
-        /// </summary>
-        public static IntPtr CreateHBitmapColorMask(Bitmap bitmap, IntPtr monochromeMask)
+     internal static Color Darker(Color color, float offset)
+            => Color.FromArgb(color.A, (int)(color.R* offset), (int) (color.G* offset), (int) (color.B* offset));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ControlPaint.cs:312]: Please correct the formatting

**Review by dreddy-work — APPROVED** [review]

**Comment by RussKie** [comment]
Is this the result of this change?
image

I'm not liking it. @merriemcgaw this looks jarring, expect customer complaints.

**Comment by merriemcgaw** [comment]
These are required accessibility change to adhere to the MAS and achieve accessibility compliance. I don't want to revert back to a less accessible default. If we hear too much heat incoming we will need to enable devs to create their apps in an inaccessible way - though it's a bit of a hard sell to approve a feature that is all about enabling reduced accessibility ☹️ . 

As @RussKie and I noted in our offline conversation - it appears that drawing that frame on Win11 does not respect the borders of the control that are rounded. We need to see what we can do to fix that part up.

**Pull request merged**